themes/classic/templates/catalog/_partials/product-prices.tpl

```diff
@@ -50,7 +50,13 @@
             {if $product.discount_type === 'percentage'}
               <span class="discount discount-percentage">{l s='Save %percentage%' d='Shop.Theme.Catalog' sprintf=['%percentage%' => $product.discount_percentage_absolute]}</span>
             {else}
-              <span class="discount discount-amount">{l s='Save %amount%' d='Shop.Theme.Catalog' sprintf=['%amount%' => $product.discount_amount]}</span>
+              <span class="discount discount-amount">
+                {if $product.reduction < $product.price_without_reduction}
+                  {l s='Save %amount%' d='Shop.Theme.Catalog' sprintf=['%amount%' => $product.discount_amount]}
```

Comment:
Can't you make the calculation into the presenter?
Don't forget we don't want any calculation into the templates